File: CCImageExplorer/CCImageExplorer/ViewController.m
//
//	iOS培训
//		传智播客 & 黑马
//		Chen Chen @ Apirl 13th, 2015
//

#import "ViewController.h"

@interface ViewController ()

/**
 *  图片视图
 */
@property (weak, nonatomic) IBOutlet UIImageView *imageView;
/**
 *  计数器标签
 */
@property (weak, nonatomic) IBOutlet UILabel *numberLabel;
/**
 *  标题标签
 */
@property (weak, nonatomic) IBOutlet UILabel *titleLabel;
/**
 *  内容标签
 */
@property (weak, nonatomic) IBOutlet UILabel *textLabel;
/**
 *  前一个按钮
 */
@property (weak, nonatomic) IBOutlet UIButton *previousButton;
/**
 *  后一个按钮
 */
@property (weak, nonatomic) IBOutlet UIButton *nextButton;
/**
 *  图片下标
 */
@property (assign, nonatomic) NSInteger index;
/**
 *  图片数组
 */
@property (strong, nonatomic) NSArray *imageArray;

@end

@implementation ViewController

#pragma mark - 初始化方法

/**
 *  图片数组惰性初始化方法
 */
- (NSArray *)imageArray
{
    if (_imageArray == nil) {
        NSString *path = [[NSBundle mainBundle] pathForResource:@"imageData" ofType:@"plist"];
        _imageArray = [NSArray arrayWithContentsOfFile:path];
    }
    return _imageArray;
}

#pragma mark - 系统方法

/**
 *  视图已经加载方法
 */
- (void)viewDidLoad
{
    [super viewDidLoad];
    [self changeImage];
}

#pragma mark - 图片方法

/**
 *  改变图片方法
 */
- (void)changeImage
{
    self.numberLabel.text = [NSString stringWithFormat:@"%ld/%ld", self.index + 1, self.imageArray.count];
    NSDictionary *imageDict = self.imageArray[self.index];
    self.imageView.image = [UIImage imageNamed:imageDict[@"no"]];
    self.titleLabel.text = imageDict[@"title"];
    self.textLabel.text = imageDict[@"text"];
    self.previousButton.enabled = (self.index != 0);
    self.nextButton.enabled = (self.index != self.imageArray.count - 1);
}

/**
 *  移动图片方法
 */
- (IBAction)moveImage:(id)sender
{
    if (sender == self.previousButton)
        self.index--;
    else if (sender == self.nextButton)
        self.index++;
    [self changeImage];
}

@end

File: CCImageExplorer-Swift/CCImageExplorer-Swift/ViewController.swift
//
//  ViewController.swift
//		CCImageExplorer
//		Chen Chen @ November 13th, 2016
//

import UIKit

import SwiftyJSON

class ViewController: UIViewController {

    /// 图片浏览视图
    private var imageExplorerView = ImageExplorerView(frame: kScreenFrame)
    /// 图片下标
    private var index: Int = 0
    /// 图片数组
    private var imageArray: [[String: Any]] = {
        var array = [[String: Any]]()
        do {
            if let data = try String(contentsOfFile: Bundle.main.path(forResource: "ImageData", ofType: "json")!).data(using: .utf8) {
                let json = JSON(data: data)
                if let jsonArray = json.arrayObject {
                    for dict in jsonArray {
                        array.append(dict as! [String: Any])
                    }
                }
            }

        } catch {
            print("读取失败")
        }
    
        return array
    }()
    
    // MARK: - 系统方法
    
    /**
     视图已经加载方法
     */
    override func viewDidLoad() {
        
        super.viewDidLoad()
        
        view = imageExplorerView
        imageExplorerView.previousButton.addTarget(self, action: #selector(moveButtonDidClick(button:)), for: .touchUpInside)
        imageExplorerView.nextButton.addTarget(self, action: #selector(moveButtonDidClick(button:)), for: .touchUpInside)
        changeImage()
    }
    
    // MARK: - 图片方法
    
    /**
     改变图片方法
     */
    private func changeImage() {
        
        imageExplorerView.numberLabel.text = String(format: "%d/%d", index + 1, imageArray.count)
        let imageDict = imageArray[index]
        imageExplorerView.imageView.image = UIImage(named: imageDict["no"] as! String)
        imageExplorerView.titleLabel.text = imageDict["title"] as? String
        imageExplorerView.textLabel.text = imageDict["text"] as? String
        
        imageExplorerView.previousButton.isEnabled = (index != 0)
        imageExplorerView.nextButton.isEnabled = (index != imageArray.count - 1)
    }
    
    /**
     移动按钮点击方法
     */
    @objc private func moveButtonDidClick(button: UIButton) {
        
        if button == imageExplorerView.previousButton {
            index -= 1
            
        } else if button == imageExplorerView.nextButton {
            index += 1
        }
        
        changeImage()
    }
    
}
